File: widgets/js/RockInnStatus.js
(function () {

	function refreshRockInn() {
		$.ajax({
			url: '/home/widget/',
			data: JSON.stringify({
				skill: 'RockInn',
				widget: 'RockInnStatus',
				func: 'getNumbers',
				param: ''
			}),
			contentType: 'application/json',
			dataType: 'json',
			type: 'POST'
		}).done(function (answer) {
			$('#RockInn_FREE').html(answer['free']);
			$('#RockInn_VISITORS').html(answer['visitors']);
		});
	}

	refreshRockInn();
	setInterval(function () {
		refreshRockInn()
	}, 1000 * 20);
})();

File: widgets/RockInnStatus.py
import sqlite3
import json

from core.base.model.Widget import Widget


class RockInnStatus(Widget):

	OPTIONS: dict = dict()

	def __init__(self, data: sqlite3.Row):
		super().__init__(data)


	def getNumbers(self):
		asDict = dict()
		asDict['visitors'], asDict['free'] = self.skillInstance.getNumbers()

		return json.dumps(asDict)

